homeassistant/components/zha/cover.py

```diff
@@ -70,40 +73,143 @@ async def async_setup_entry(
-        """Init this sensor."""
+    def __init__(
+        self,
+        unique_id: str,
+        zha_device: ZHADevice,
+        cluster_handlers: list[ClusterHandler],
+        **kwargs: Any,
+    ) -> None:
+        """Init this cover."""
         super().__init__(unique_id, zha_device, cluster_handlers, **kwargs)
-        self._cover_cluster_handler = self.cluster_handlers.get(CLUSTER_HANDLER_COVER)
-        self._current_position = None
-        self._tilt_position = None
+        cluster_handler = self.cluster_handlers.get(CLUSTER_HANDLER_COVER)
+        assert cluster_handler
+        self._cover_cluster_handler: WindowCoveringClusterHandler = cast(
+            WindowCoveringClusterHandler, cluster_handler
+        )
+        if self._cover_cluster_handler.window_covering_type:
+            self._attr_device_class: CoverDeviceClass | None = (
+                ZCL_TO_COVER_DEVICE_CLASS.get(
+                    self._cover_cluster_handler.window_covering_type
+                )
+            )
+        self._attr_supported_features: CoverEntityFeature = (
+            self._determine_supported_features()
+        )
+        self._target_lift_position: int | None = None
+        self._target_tilt_position: int | None = None
+        self._determine_initial_state()
+
+    def _determine_supported_features(self) -> CoverEntityFeature:
+        """Determine the supported cover features."""
+        supported_features: CoverEntityFeature = (
+            CoverEntityFeature.OPEN
+            | CoverEntityFeature.CLOSE
+            | CoverEntityFeature.STOP
+            | CoverEntityFeature.SET_POSITION
+        )
+        if (
+            self._cover_cluster_handler.window_covering_type
+            and self._cover_cluster_handler.window_covering_type
+            in (
+                WCT.Shutter,
+                WCT.Tilt_blind_tilt_only,
+                WCT.Tilt_blind_tilt_and_lift,
+            )
+        ):
+            supported_features |= CoverEntityFeature.SET_TILT_POSITION
+            supported_features |= CoverEntityFeature.OPEN_TILT
+            supported_features |= CoverEntityFeature.CLOSE_TILT
+            supported_features |= CoverEntityFeature.STOP_TILT
+        return supported_features
+
+    def _determine_initial_state(self) -> None:
+        """Determine the initial state of the cover."""
+        if (
+            self._cover_cluster_handler.window_covering_type
+            and self._cover_cluster_handler.window_covering_type
+            in (
+                WCT.Shutter,
+                WCT.Tilt_blind_tilt_only,
+                WCT.Tilt_blind_tilt_and_lift,
+            )
+        ):
+            self._determine_state(self.current_cover_tilt_position)
+            if (
+                self._cover_cluster_handler.window_covering_type
+                == WCT.Tilt_blind_tilt_and_lift
+            ):
+                state = self._state
+                self._determine_state(self.current_cover_position)
```

Comment:
... because directly below, it gets called with cover position (aka lift).
(where the default `is_lift_update=True` makes sense)